modules/apps/login/login-authentication-openid-connect-web/src/main/java/com/liferay/login/authentication/openid/connect/web/internal/portlet/action/OpenIdConnectLoginRequestMVCRenderCommand.java

```diff
@@ -64,11 +68,42 @@ public String render(
 			return "/login.jsp";
 		}
 
-		httpServletRequest.setAttribute(
-			OpenIdConnectWebKeys.OAUTH_CLIENT_ENTRIES,
+		List<OAuthClientEntry> oauthClientEntries =
 			_oAuthClientEntryLocalService.
 				getAuthServerWellKnownURISuffixOAuthClientEntries(
-					themeDisplay.getCompanyId(), "openid-configuration"));
+					themeDisplay.getCompanyId(), "openid-configuration");
+
+		if (oauthClientEntries.size() == 1) {
+			String portletURLString = PortletURLBuilder.createActionURL(
+				renderResponse
+			).setActionName(
+				OpenIdConnectWebKeys.OPEN_ID_CONNECT_REQUEST_ACTION_NAME
+			).setParameter(
+				"oAuthClientEntryId",
+				oauthClientEntries.get(
+					0
+				).getOAuthClientEntryId()
+			).setParameter(
+				"saveLastPath", false
+			).buildString();
+
+			HttpServletResponse httpServletResponse =
+				_portal.getHttpServletResponse(renderResponse);
+
+			try {
+				httpServletResponse.sendRedirect(portletURLString);
```

Comment:
Hi @nikolettabuza . I am not sure this is a good idea to issue a redirect response from the portlet's render phase. With portlets you don't know what other portlets are being rendered on the same page. So none of them should hi-jack the HTTP response in this way.

Reply:
I've maked this search across portal code and we have several examples of sendRedirect in MVCRenderCommands, some of them adds 
return MVCRenderConstants.MVC_PATH_VALUE_SKIP_DISPATCH; 
after this redirection.